.github/workflows/ci.yml

```diff
@@ -712,3 +725,166 @@ jobs:
+            LATEST_TAG=$(jq -r '.[].name' __response.json | grep -E ^$ESCAPED_REF_NAME\.[0-9]+$ | sort -V -r | head -n 1)
+            echo "LATEST_TAG is $LATEST_TAG"
+            echo "latest_tag=$LATEST_TAG" >> $GITHUB_OUTPUT
+            if ! [[ $LATEST_TAG =~ ([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
+              echo "Not patch releasing because cannot find a matching semver tag on the branch"
+              DO_RELEASE=0
+            else
+              MAJOR=${BASH_REMATCH[1]}
+              MINOR=${BASH_REMATCH[2]}
+              PATCH=${BASH_REMATCH[3]}
+              NEXT_TAG="$MAJOR.$MINOR.$((PATCH+1))"
+              echo "NEXT_TAG is $NEXT_TAG"
+              echo "next_tag=$NEXT_TAG" >> $GITHUB_OUTPUT
+            fi
+          fi
+
+          # Check if there is anything relevant to release using GitHub API
+          # The API results include all merged pull-requests, not commits
+          # Pull-requests prefixed with MNT or DOC will not be considered relevant for releasing
+          if [[ $DO_RELEASE == "1" ]]; then
+            # Check on github release notes api if there's anything worth releasing
+            # https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X POST https://api.github.com/repos/$GITHUB_REPOSITORY/releases/generate-notes \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            -d @- << EOF
+            {
+              "tag_name": "$NEXT_TAG",
+              "target_commitish": "$LATEST_LOCAL_SHA",
+              "previous_tag_name": "$LATEST_TAG"
+            }
+          EOF
+            # ^ The EOF line above intentionally indented like it is
+            )
+            if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to fetch release notes - HTTP response code was $RESP_CODE"
+              cat __response.json
+              exit 1
+            fi
+            # Get raw body for text parsing
+            jq -r '.body' __response.json > __body.json
+            # Parse release notes one line at a time
+            IS_PRS_SECTION=0
+            HAS_THINGS_TO_RELEASE=0
+            while IFS="" read -r line || [[ -n $line ]]; do
+              echo "$line"
+              if [[ $IS_PRS_SECTION == "0" ]]; then
+                if [[ "$line" == "## What's Changed" ]]; then
+                  IS_PRS_SECTION=1
+                fi
+              else
+                if [[ "$line" =~ ^\*[[:space:]](.+?)$ ]]; then
+                  PR_TITLE=${BASH_REMATCH[1]}
+                  if ! [[ $PR_TITLE =~ ^(MNT|DOC) ]]; then
+                    HAS_THINGS_TO_RELEASE=1
+                  fi
+                else
+                  IS_PRS_SECTION=0
+                fi
+              fi
+            done < __body.json
```

Comment:
Using release notes means we're only looking at the names of PRs - which frankly are irrelevant. Consider for example a PR named "fix broken builds" where the commit message is "MNT blah blah broken build". The release notes will have "fix broken builds" so this will say "yay that's a fix!" But the commit is MNT.

Use https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits instead. This gives us some nice json where we can check each commit's message. You can cleanly compare from the latest tag to the current commit sha. See for example:
https://api.github.com/repos/silverstripe/silverstripe-framework/compare/4.13.8...5303e08d34c80348514a6c1d90815450b2518921

This API also gives us more information (e.g. about the actual files changes) in case we wanted to do some more specific stuff with this later on.

Reply:
Have updated to use commits - have tested that it works correctly with a [MNT](https://github.com/emteknetnz/silverstripe-config/actions/runs/5420959807) followed by a [ENH](https://github.com/emteknetnz/silverstripe-config/actions/runs/5420978723) commit (didn't use PR's for these hence [auto release notes](https://github.com/emteknetnz/silverstripe-config/releases/tag/2.0.4) don't mention them)